.github/workflows/tests-unit.yml

```diff
@@ -0,0 +1,22 @@
+name: Run Automated Tests
+on:
+  workflow_call
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v2
+      - name: Set up JDK 14
+        uses: actions/setup-java@v1
+        with:
+          java-version: 14
+      - name: Cache Maven packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+      - name: Run unit tests
+        run: mvn -B test --file pom.xml
```

Comment:
And thanks a ton @grant-mcs it was a _breeze_ to copy and paste from playground.

Reply:
@markpatton should this be `mvn test` and `mvn verify`?

Reply:
We will capture `verify` separately as an integration test and not run as often as it's much slower